File: src/main/java/io/github/gravetii/scene/start/StartControlsComponent.java
package io.github.gravetii.scene.start;

import io.github.gravetii.scene.FxComponent;
import io.github.gravetii.scene.menu.MenuBarComponent;

import javafx.scene.layout.VBox;

public class StartControlsComponent extends FxComponent<StartControlsController, VBox> {
  private final MenuBarComponent ref;

  public StartControlsComponent(MenuBarComponent ref) throws Exception {
    super("startControls.fxml");
    this.ref = ref;
    this.create();
  }

  @Override
  protected StartControlsController createController() {
    return new StartControlsController(this.ref.getController());
  }

  @Override
  protected VBox createNode() throws Exception {
    return this.loadNode();
  }
}

File: src/main/kotlin/io/github/gravetii/scene/start/StartControlsComponent.kt
package io.github.gravetii.scene.start

import io.github.gravetii.controller.StartControlsController
import io.github.gravetii.scene.FxComponent
import io.github.gravetii.scene.menu.MenuBarComponent
import javafx.scene.layout.VBox

class StartControlsComponent(ref: MenuBarComponent) :
    FxComponent<StartControlsController, VBox>("startControls.fxml") {

    override val controller: StartControlsController = StartControlsController(ref.controller)

    override val node: VBox = loadNode()

}